### PR TITLE
[Model] Add ColQwen3.5 4.5B support

### PR DESCRIPTION
**Pull request by athrael-soju**
## Purpose

Add support for **ColQwen3.5** late interaction model for multi-modal document retrieval and reranking.

[ColQwen3.5](https://huggingface.co/athrael-soju/colqwen3.5-4.5B) extends the [ColPali](https://arxiv.org/abs/2407.01449) architecture using the Qwen3.5 hybrid backbone (linear + full attention). It produces per-token L2-normalized embeddings and uses ColBERT-style MaxSim scoring for visual document retrieval.

This follows the same pattern as the existing ColQwen3 support (#34398).

### Changes

**New files:**
- `vllm/model_executor/models/colqwen3_5.py` — Model implementation (`ColQwen3_5Model`) extending `Qwen3_5ForConditionalGeneration` with ColBERT projection head
- `tests/models/multimodal/pooling/test_colqwen3_5.py` — Tests for token embedding, MaxSim scoring, and relevance ordering
- `examples/pooling/score/colqwen3_5_rerank_online.py` — Online reranking example

**Modified files:**
- `vllm/model_executor/models/registry.py` — Register `ColQwen3_5` architecture
- `tests/models/registry.py` — Add test model entry
- `docs/models/supported_models.md` — Add to embedding models table
- `docs/models/pooling_models.md` — Add usage documentation section

## Test Plan

```bash
pytest tests/models/multimodal/pooling/test_colqwen3_5.py -v
```

Tests cover:
- `test_colqwen3_5_token_embed` — Verifies per-token embedding shape `(seq_len, 320)` and L2 normalization
- `test_colqwen3_5_late_interaction_scoring` — Verifies MaxSim scores via `/score` match manual computation
- `test_colqwen3_5_relevance_ordering` — Verifies relevant documents score higher than irrelevant ones

## Test Result

All 3 tests pass locally (WSL2, NVIDIA GPU):

```
tests/models/multimodal/pooling/test_colqwen3_5.py::test_colqwen3_5_token_embed[half-athrael-soju/colqwen3.5-4.5B] PASSED
tests/models/multimodal/pooling/test_colqwen3_5.py::test_colqwen3_5_late_interaction_scoring[half-athrael-soju/colqwen3.5-4.5B] PASSED
tests/models/multimodal/pooling/test_colqwen3_5.py::test_colqwen3_5_relevance_ordering[half-athrael-soju/colqwen3.5-4.5B] PASSED

3 passed in 462.24s
```

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [x] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/model_executor/models/colqwen3_5.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+                if self.embed_dim is None and "weight" in name:
+                    self.embed_dim = weight.shape[0]
+                    has_bias = any("bias" in n for n, _ in proj_weights)
+                    self.custom_text_proj = nn.Linear(
+                        self._proj_hidden_size,
+                        self.embed_dim,
+                        bias=has_bias,
+                        dtype=model_dtype,
+                    )
+                    self.custom_text_proj.to(model_device)
```
[the remark this answers is not in the document]
Comment (athrael-soju) [reply vllm/model_executor/models/colqwen3_5.py]: Addressed

File: vllm/model_executor/models/colqwen3_5.py [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+        if not isinstance(hidden_states, torch.Tensor):
+            return hidden_states  # type: ignore
+
+        proj_dtype = self.custom_text_proj.weight.dtype  # type: ignore
+        if hidden_states.dtype != proj_dtype:
+            hidden_states = hidden_states.to(proj_dtype)
+
+        # Project to embedding dimension and L2 normalize
+        proj = self.custom_text_proj(hidden_states)  # type: ignore
+        return torch.nn.functional.normalize(proj, p=2, dim=-1)
```
Comment (noooop) [inline vllm/model_executor/models/colqwen3_5.py]: Normalization should be done by pooler

Comment (athrael-soju) [reply vllm/model_executor/models/colqwen3_5.py]: Addressed!

File: vllm/model_executor/models/registry.py
Comment (noooop) [inline vllm/model_executor/models/registry.py:263]: Please move ColQwen3_5 and ColPaliForRetrieval to _LATE_INTERACTION_MODELS.

File: tests/models/registry.py [hunk 3]
```diff
@@ -597,6 +597,9 @@ def check_available_online(
         "TIGER-Lab/VLM2Vec-Full", trust_remote_code=True
     ),
     "Qwen2VLForConditionalGeneration": _HfExamplesInfo("MrLight/dse-qwen2-2b-mrl-v1"),
+    "ColQwen3_5": _HfExamplesInfo(
+        "athrael-soju/colqwen3.5-4.5B", trust_remote_code=True
+    ),
     "SiglipModel": _HfExamplesInfo("google/siglip-base-patch16-224"),
     "PrithviGeoSpatialMAE": _HfExamplesInfo(
```
Comment (noooop) [inline tests/models/registry.py:601]: Please move ColQwen3_5 to _LATE_INTERACTION_EXAMPLE_MODELS

Comment (athrael-soju) [reply tests/models/registry.py:601]: Done!

File: vllm/model_executor/models/registry.py
Comment (athrael-soju) [reply vllm/model_executor/models/registry.py:263]: Done!

**Comment by noooop** [comment]
This failure has still not been fixed.

**Review by noooop — APPROVED** [review]
great to have this
